Detectors/FIT/benchmark/process.py

```diff
@@ -0,0 +1,368 @@
+
+elif title.find(digitization) == 0:
+    command=digitization
+    print("You have monitored o2 digitization.\n")
+    logfilename = 'o2digi.log'
+
+else :
+    print("I do not know this type of simulation.\n")
+    exit(1)
+
+#################################################
+#                                               #
+#       Extract the PIDs from logfile           #
+#                                               #
+#################################################
+
+if command==simulation: # True if you typed o2-sim
+    
+    try:
+        # open o2sim.log file name
+        with open(logfilename) as logfile:
+            # read and save the first 6 lines in o2sim.log
+            loglines = [next(logfile) for line in range(6)]
+
+    #    print("*******************************\n")
+    #    print("Driver application PID is: ", driver)
+
+        # find the PID for the event generator (o2-sim-primary-..)
+        eventgenerator_line = re.search('Spawning particle server on PID (.*); Redirect output to serverlog\n',loglines[3])
+        event_gen = float(eventgenerator_line.group(1))
+    #    print("Eventgenerator PID is: ", event_gen)
+
+        # find the PID for sim worker 0 (o2-sim-device-runner)
+        sim_worker_line = re.search('Spawning sim worker 0 on PID (.*); Redirect output to workerlog0\n',loglines[4])
+        sim_worker = float(sim_worker_line.group(1))
+    #    print("SimWorker 0 PID is: ", sim_worker)
+
+        # find the PID for the hitmerger (o2-sim-hitmerger)
+        hitmerger_line = re.search('Spawning hit merger on PID (.*); Redirect output to mergerlog\n',loglines[5])
+        hit_merger = float(hitmerger_line.group(1))
+    #    print("Hitmerger PID is: ", hit_merger, "\n")
+    #    print("*******************************\n")
+
+        # find the number of simulation workers
+        n_workers = int(re.findall('Running with (\\d+)', loglines[1])[0])
+
+        # save into a list
+        pid_names = ['driver','event gen','sim worker 0','hit merger']
+        pid_vals = [driver,event_gen,sim_worker,hit_merger]
+
+        # append pid names for remaining workers
+        for i in range(n_workers-1):
+            pid_names.append(f"sim worker {i+1}")
+        
+        no_log = False
+    
+    except IOError:
+        print("There exists no o2sim.log..")
+        print("No details of devices will be provided.")
+        no_log = True
+
+elif command==digitization: # True if you typed o2-sim-digitizer-workflow
+
+    try:
+        # open o2digi.log file name
+        with open(logfilename) as logfile:
+
+            # save the first 100 lines in o2digi.log
+            loglines = [next(logfile) for line in range(100)]
+
+        # declare list for PID numbers and names
+        pid_vals = []
+        pid_names = []
+
+        # loop through lines to find PIDs
+        for line_num,line in enumerate(loglines)
+            pid_line = re.findall('Starting (\\w.+) on pid (\\d+)', loglines[line])
```

Comment:
Line 135:` pid_line = re.findall('Starting (\\w.+) on pid (\\d+)', loglines[line])` has to be changed to 
`pid_line = re.findall('Starting (\\w.+) on pid (\\d+)',line)`

Reply:
Done